x/incentives/keeper/grpc_query.go

```diff
@@ -143,6 +143,28 @@ func (q Querier) UpcomingGauges(goCtx context.Context, req *types.UpcomingGauges
 	return &types.UpcomingGaugesResponse{Data: gauges, Pagination: pageRes}, nil
 }
 
+func (q Querier) UpcomingGaugesPerDenom(goCtx context.Context, req *types.UpcomingGaugesPerDenomRequest) (*types.UpcomingGaugesPerDenomResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
```

Comment:
Nit: I know the other queries don't do this, which we need to address IMO, but we should have an explicit `nil` check here for `req`.

i.e.
```go
if req == nil {
  	if req == nil {
		return nil, status.Error(codes.InvalidArgument, "empty request")
	}
}

	if req.Denom == "" {
		return nil, status.Error(codes.InvalidArgument, "invalid denom")
	}

```

Reply:
Granted, we probably shouldn't use `InvalidArgument` for `req` being nil, but rather something else.

Reply:
Hrmm, lets make a second issue for this? It'd be nice if theres something we can do one layer up to save if `if req == nil` in every single querier logic call

Reply:
Sounds good! If you create it, you can assign it to me and I'll knock it out 👍

Reply:
Issue made #1224